### PR TITLE
make validator i18n friendly

### PR DESCRIPTION
**Pull request by maxchursin**

### CODE REVIEW

File: AndroidManifest.xml [hunk 1]
```diff
@@ -1,10 +1 @@
...
-    package="com.mobsandgeeks.saripaar"
-    android:versionCode="1"
-    android:versionName="1.0">
-
-    <uses-sdk android:minSdkVersion="4" android:targetSdkVersion="15" />
-
-    <application></application>
-
-</manifest>
+<manifest package="com.mobsandgeeks.saripaar"/>
```
Comment (objcode) [inline AndroidManifest.xml]: There was an issue last release with not specifying `minSdkVersion` in libraries causing extra permissions to be added. Is this safe?

File: src/com/mobsandgeeks/saripaar/AnnotationRuleFactory.java [hunk 2]
```diff
@@ -162,13 +166,17 @@
         if (minLength > 0) {
             rules.add(Rules.minLength(null, minLength, textRule.trim()));
         }
-        if (maxLength != Integer.MAX_VALUE) {
+        if (maxLength < Integer.MAX_VALUE) {
```
Comment (objcode) [inline src/com/mobsandgeeks/saripaar/AnnotationRuleFactory.java]: Does this change functionality?

**Comment by objcode** [comment]
:+1: pending checking on `minSdkVersion`

File: src/main/res/values/attrs.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <declare-styleable name="DefaultErrorStingsTheme">
+        <attr name="DefaultErrorStingsStyle" format="reference" />
```
Comment (objcode) [inline src/main/res/values/attrs.xml]: s/DefaultErrorStingsStyle/DefaultErrorStringsStyle/g :)

sting-feydrautha

File: AndroidManifest.xml (hunk 1 above)
Comment (maxchursin) [inline AndroidManifest.xml]: oh, I wasn't aware of the causes of  that issue

File: src/com/mobsandgeeks/saripaar/AnnotationRuleFactory.java (hunk 2 above)
Comment (maxchursin) [inline src/com/mobsandgeeks/saripaar/AnnotationRuleFactory.java]: not at all, it's conceptually seems more correct

File: AndroidManifest.xml (hunk 1 above)
Comment (maxchursin) [inline AndroidManifest.xml]: it actually specified in the gradle script, and not needed here

Comment (objcode) [inline AndroidManifest.xml]: :ok:

**Pull request merged**